src/lib.rs

```diff
@@ -193,7 +195,34 @@ where
                 return Err(PollWfError::ShutDown);
             }
 
-            match abort_on_shutdown!(self, poll_workflow_task, task_queue.to_owned()) {
+            let shutdownfut = async {
+                loop {
+                    self.shutdown_notify.notified().await;
+                    if self.shutdown_requested.load(Ordering::SeqCst) {
+                        break;
+                    }
+                }
+            };
+            let pa_fut = async {
+                loop {
+                    self.pending_activation_notify.notified().await;
+                    break;
+                }
+            };
+            let poll_result_future = self
+                .server_gateway
+                .poll_workflow_task(task_queue.to_owned());
+            let selected_f = tokio::select! {
+                _ = shutdownfut => {
+                    Err(ShutdownErr.into())
+                }
+                _ = pa_fut => {
```

Comment:
What happens to the poll request in this case?
Is it cancelled? Is it ignored?
We probably don't want to create a new request if we got interrputed.

Reply:
Yeah it's a good question, was thinking about it after I wrote this. I'll see if I can figure out a way to test that while I finish this. Ideally, it doesn't really matter, because we spit out the PA so fast that when we go back to requesting it'll kinda be like nothing ever happened... but perhaps we need to keep the polling future around in order to ensure that behavior. I'll play with it.

Reply:
I think this opens up too big of a can of worms to deal with in this PR. Ideally the request should actually be cancelled I think. Keeping it around raises too many questions like "what if the next poll is for a different task queue" or "how long should we buffer the response" and so on. I opened https://github.com/temporalio/sdk-core/issues/91

Reply:
I agree, we should discard initial poll as from the lang's perspective it polled and got an activation, it shouldn't care if core made a request to the server or not in order to get it.